powershell/VstsTaskSdk/InputFunctions.ps1

```diff
@@ -158,60 +160,121 @@ function Get-EnvVariable {
         }
     }
 
-    if ($item -and $item.value) {
-        $value = $item.value
-        Write-Verbose "$($path): '$value'"
+    Get-FinalValue -Value $item.Value @PSBoundParameters
+}
+
+function Get-SecureInput {
+    [CmdletBinding(DefaultParameterSetName = 'Require')]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Name,
+        [Parameter(Mandatory = $true)]
+        [string]$Path,
+        [Parameter(ParameterSetName = 'Require')]
+        [switch]$Require,
+        [Parameter(ParameterSetName = 'Default')]
+        [object]$Default,
+        [switch]$AsBool,
+        [switch]$AsInt)
+
+    # Attempt to get the secure variable.
+    $value = $null
+    if ($psCredential = $script:secureInputs[$Path]) {
+        $value = $psCredential.GetNetworkCredential().Password
+    } elseif (!$script:nonInteractive) {
+        # The value wasn't found. Prompt for the value if running in interactive dev mode.
+        $value = Read-Host -Prompt $Name
+        if ($value) {
+            $script:secureInputs[$Path] = New-Object System.Management.Automation.PSCredential(
+                $Path,
+                (ConvertTo-SecureString -String $value -AsPlainText -Force))
+        }
+    }
+
+    Get-FinalValue -Value $value @PSBoundParameters
+}
+
+function Get-FinalValue {
```

Comment:
This new (private) function contains the common code used by both the existing Get-EnvVariable function (called by Get-TaskVariable) and the new Get-SecureInput function (called by Get-Endpoint/Get-Input).